### PR TITLE
feat: add CI checks

### PR DESCRIPTION
**Pull request by sujithsomraaj**
solves #46

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+
+      - name: Build the contracts
+        run: forge build --build-info
+
+      - name: Run Slither
+        uses: crytic/slither-action@v0.3.0
+        with:
+          fail-on: none
```
Comment (ermyas) [inline .github/workflows/ci.yml:84]: I'd suggest we fail on at least a 'high' finding

File: src/MultiMessageSender.sol [hunk 2]
```diff
@@ -202,32 +213,33 @@ contract MultiMessageSender {
...
 
-        bool[] memory adapterSuccess = new bool[](adapterLength);
+        v.adapterSuccess = new bool[](v.adapterLength);
 
-        for (uint256 i; i < adapterLength;) {
-            IBridgeSenderAdapter bridgeAdapter = IBridgeSenderAdapter(adapters[i]);
+        for (uint256 i; i < v.adapterLength;) {
+            IBridgeSenderAdapter bridgeAdapter = IBridgeSenderAdapter(v.adapters[i]);
 
             address mmaReceiver = gac.getMultiMessageReceiver(_dstChainId);
```
Comment (ermyas) [inline src/MultiMessageSender.sol:244]: we can move lines 244-248 outside the `for` loop

File: test/Setup.t.sol [hunk 3]
```diff
@@ -3,18 +3,18 @@ pragma solidity >=0.8.9;
...
+import "forge-std/console.sol";
 
 /// @dev imports from Pigeon Helper (Facilitate State Transfer Mocks)
 import {WormholeHelper} from "pigeon/wormhole/WormholeHelper.sol";
 import {AxelarHelper} from "pigeon/axelar/AxelarHelper.sol";
 
 /// local imports
-import {WormholeSenderAdapter} from "../src/adapters/wormhole/WormholeSenderAdapter.sol";
-import {WormholeReceiverAdapter} from "../src/adapters/wormhole/WormholeReceiverAdapter.sol";
+import {WormholeSenderAdapter} from "src/adapters/Wormhole/WormholeSenderAdapter.sol";
```
Comment (Dominator008) [inline test/Setup.t.sol:13]: Should we unify on the casing? I think folder names should be `wormhole` and `celer`.

Comment (Dominator008) [reply test/Setup.t.sol:13]: SG

**Review by Dominator008 — APPROVED** [review]

File: test/Setup.t.sol (hunk 3 above)
Comment (ermyas) [reply test/Setup.t.sol:13]: added this to the list in issue #43

**Pull request merged**